HA-flow history: add dumps, fix task IDs for saving an action to history

dmitrii-beliakov opened the pull request:
- [x] After this fix, the action of de-allocating resources will be visible in the list of actions under the history event of deleting HA-flow on.
- [x]  fix also allows to use simple flow history methods in HA-flow FSMs (but arguably this is a temporary solution).
- [x] add dumps before and after to the HA flow reroute event.
- [x] add a dump before to the HA flow delete event.

closes #5425

yuliiamir commented:
After adding a new logic, a side issue has been caught for the reroute event: 
```
Column ‘flow_event_id’ cannot be null
Unhandled exception in org.openkilda.wfm.topology.history.bolts.HistoryBolt
```
TC to reproduce the issue: 
_History records are created during link unsuccessful rerouting with root cause details and can be retrieved with or without timeline_

dmitrii-beliakov commented:
I added support of using simple flow history methods that are used in HA-flow FSMs via base classes dedicated to simple flows. The errors will be saved as HA-flow history events. At the same time a warning will be logged in ELK.
I extended this logging for cases of using saving actions of simple flow history from HA-flow FSM. I added a TODO and will create an issue for a proper solution; in general this not correct to save simple flow history from HA-flow, because these history records have different format and are read from different API endpoints.

yuliiamir commented:
After discussion, it was decided to add changes in the scope of fixing the issue https://github.com/telstra/open-kilda/issues/5425

dmitrii-beliakov commented:
I added a dump state after for HA flow delete operation. For reroute I added before and after. With reroute it is more complicated. Rerouting could have three endings: completed with success, completed without rerouting, ended with an error. Dumps are saved for the first two, but the latest don't have the dump after.

There is a potential concurrency issue. And at the same time the problem of definition of the dump. For these operation a dump is a copy of HA flow at the moment of saving. FSMs don't work fully in a blocking way, there are some actions still in progress, while a concurrent event could possibly already start changing the flow. So, dumps for these operations should be treated as "it is true that the flow had this state at that time". They don't guarantee anything else. Other operation could have dumps implemented differently: some FSMs have a target flow saved at earlier point in time than the dump is saved. It means that the time of a dump after could overlap with the next operation. However, this is unlikely to happen on average case in production environment.

niksv reviewed (approved):
LGTM
